### PR TITLE
(dev/drupal/10) Make civicrm-version.php self-sufficient to fetch Civi version and CMS name

### PR DESCRIPTION
**Pull request by monishdeb**
Overview
----------------------------------------
This PR is about making ```civicrm-version.php``` self-sufficient to get Civi version and CMS name, which means we don't need CRM_Core_CodeGen_Version to generate the same file. 

Technical Details
----------------------------------------
This patch makes following additional improvement:
1. Doesn't need to rely on ```CRM_Core_CodeGen_Version``` to generate version file.
2. Makes ```civicrm-version.php``` independent to fetch Civi Version and CMS.
3. ```civicrm-version.php``` is now one of the core file whereas ```CRM/Core/CodeGen/Version.php``` is not.
4. Now ```civicrm-version.php``` introduces two new functions to get Civi version and CMS name respectively. See the changes made in CRM_Utils_System::version() as an example.

### CODE REVIEW

**Comment by monishdeb** [comment]
ping @totten so far I have tested in Drupal7/8, Backdrop, Wordpress and Joomla. Is there anything missing? 

With this change now I can do this to
```patch
diff --git a/drush/civicrm.drush.inc b/drush/civicrm.drush.inc
index d2b5e3c..dca10e7 100644
--- a/drush/civicrm.drush.inc
+++ b/drush/civicrm.drush.inc
@@ -492,8 +492,7 @@ function _civicrm_generate_settings_file($dbuser, $dbpass, $dbhost, $dbname, $mo
 
   if (file_exists("$crmPath/civicrm-version.php")) {
     require_once "$crmPath/civicrm-version.php";
-    $v = civicrmVersion();
-    $cms = $v['cms'];
+    $cms = findCMS();
   }
 
   $params = array(
```

**Comment by colemanw** [comment]
I have to say I am confused about the difference between `civicrm-version.php` and `version.xml`. Both files appear to be used, yet they are redundant.

Maybe that's outside the scope of this PR but it's been bugging me.

**Comment by monishdeb** [comment]
@colemanw I agree as per the file name, it seems redundant as if they only hold the Civi version. But civicrm-version.php is a php script which as per the current changes:
1. Find CMS type by guessing the current path or environment variable.
2. Fetch the Civi version from civicrm.xml file
3. As per the change this php script has some useful function which we can reuse in other places like civicrm.config.php to get the CMS type and root path. 

Whereas version.xml just holds the version. Now we can remove version.xml and directly hardcode the version value in php script and I didn't find any unintended regression or side effect, i.e. 
```
function civicrmVersion() {
   return [
     'version' => '5.2.alpha2',
     'cms' => findCMS(),
   ];
 }

File: civicrm-version.php [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+   */
+  public static function findVersion() {
+    $verFile = implode(DIRECTORY_SEPARATOR, [dirname(__FILE__), 'xml', 'version.xml']);
+    if (file_exists($verFile)) {
+      $str = file_get_contents($verFile);
+      $xmlObj = simplexml_load_string($str);
+      return (string) $xmlObj->version_no;
+    }
+
+    trigger_error("Unkown version", E_USER_ERROR);
```
Comment (mattwire) [inline civicrm-version.php]: "Unkown" -> "Unknown"

**Comment by totten** [comment]
@monishdeb I did a lot of testing (`r-run`). There were various issues (which I patched while testing), so I've extended the PR and resubmitted as #12113. If you're OK with the subsequent revisions, then I'm OK with merging.

**Pull request closed without merging**